src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,10 +1753,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
-                return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
-            } else {
+                return result;
```

Comment:
refresh token request doesn't involve with UI. I'm not returning token failure as successful case, I'm just returning the AuthenticationResult(could contain token or error), caller will check if it contains token, if it does, callback on success will be called. If it contains error, caller will check if it's non-silent request, and if it is, will start the interactive flow. The whole purpose of this method is purely just trying to get token with refresh token.